src/core/primitives.js

```diff
@@ -199,6 +199,38 @@ class Dict {
     this._map.set(key, value);
   }
 
+  setIfNotExists(key, value) {
+    if (!this.has(key)) {
+      this.set(key, value);
+    }
+  }
+
+  setNumber(key, value) {
+    if (typeof value === "number") {
+      this.set(key, value);
+    }
+  }
+
+  setArray(key, value) {
+    if (Array.isArray(value) || ArrayBuffer.isView(value)) {
+      this.set(key, value);
+    }
+  }
+
+  setSomething(key, value) {
+    if (value) {
+      this.set(key, value);
+    }
+  }
+
+  setName(key, value) {
```

Comment:
Could we perhaps use slightly more "consistent" names for these new methods?
```suggestion
  setIfNumber(key, value) {
    if (typeof value === "number") {
      this.set(key, value);
    }
  }

  setIfArray(key, value) {
    if (Array.isArray(value) || ArrayBuffer.isView(value)) {
      this.set(key, value);
    }
  }

  setIfDefined(key, value) {
    if (value) {
      this.set(key, value);
    }
  }

  setIfName(key, value) {
```